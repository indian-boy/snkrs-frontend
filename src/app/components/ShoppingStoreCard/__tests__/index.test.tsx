import { render } from '@testing-library/react';
import * as React from 'react';
import { ShoppingStoreCard } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const shoppingStoreData = {
  id: 1,
  placeName: 'Avenida paulista',
  address: {
    latitude: 0,
    longitude: 0,
    location: 'Avenida Paulista, 1227 - Bela Vista São Paulo - SP, 01311-200',
  },
  availability: 4,
  serviceHours: 'Segunda a Sábado 10h às 22h | Domingo 11h às 20h',
};

describe('<ShoppingStoreCard  />', () => {
  it('should match snapshot', () => {
    const shoppingStoreCard = render(
      <ShoppingStoreCard {...shoppingStoreData} />,
    );

    expect(shoppingStoreCard.container.firstChild).toMatchSnapshot();
  });
});
