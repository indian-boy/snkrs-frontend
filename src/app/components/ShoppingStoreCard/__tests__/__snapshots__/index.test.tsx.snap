// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingStoreCard  /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  white-space: break-spaces;
  font-weight: 500;
  font-size: 1.5rem;
  line-height: 1.2rem;
  color: #1A1A1A;
}

.c2 {
  white-space: nowrap;
  font-weight: 500;
  font-size: 1.125rem;
  color: #1A1A1A;
}

.c3 {
  background-color: transparent;
  font-style: normal;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
  cursor: pointer;
  outline: none;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  color: #1A1A1A;
  margin-top: 0.5rem;
  padding: 0;
}

.c3:focus,
.c3:active {
  outline: none;
}

.c4 {
  background-color: #FFFFFF;
  margin: 1rem 0;
  border-radius: 0.625rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 1.5rem;
  gap: 0.5rem;
}

.c5 {
  font-style: normal;
  font-weight: normal;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #737373;
}

.c6 {
  font-style: normal;
  font-weight: bold;
  font-size: 0.75rem;
  line-height: 1.5rem;
  color: #006611;
}

<div
  class="styles__Wrapper-sc-103prks-0"
>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      Avenida paulista
    </span>
    <span
      class="c2"
    >
      1km
    </span>
  </div>
  <button
    class="c3"
    role="showShoppingStoreOnMap"
    type="button"
  >
    <svg>
      pin_location.svg
    </svg>
    components.shoppingStoreCard.seeOnMapLabel
    Ver no mapa
  </button>
  <div
    class="c4"
  >
    <span
      class="c5"
    >
      Avenida Paulista, 1227 - Bela Vista São Paulo - SP, 01311-200
    </span>
    <span
      class="c5"
    >
      components.shoppingStoreCard.service
      Atendimento
      :
      <br />
      Segunda a Sábado 10h às 22h | Domingo 11h às 20h
    </span>
    <span
      class="c6"
    >
      components.shoppingStoreCard.availability
      Disponível em {{count}} dias úteis
    </span>
  </div>
</div>
`;
